engine/runtime-compiler/src/main/scala/org/enso/compiler/data/BindingsMap.scala

```diff
@@ -595,15 +592,7 @@ object BindingsMap {
     def allFieldsDefaulted: Boolean = arguments.forall(_.hasDefaultValue)
   }
 
-  case class Argument(
-    name: String,
-    hasDefaultValue: Boolean,
-    typReference: Reference[Expression]
-  ) {
-    def typ(): Option[Expression] = Option(
-      typReference.get(classOf[Expression])
-    )
-  }
```

Comment:
This removes the infamous cycle from the IR - the `Expression` holding the type of the constructor is no longer needed.

That's because we process this type `Expression` in `StaticModuleScopeAnalysis` and save it in `StaticModuleScope` metadata as an already processed `TypeRepresentation` - so it is no longer IR that is stored inside of metadata, but a simpler data structure representing our types.